plugins/dell-dock/fu-dell-dock-plugin.c

```diff
@@ -38,72 +46,93 @@ fu_dell_dock_plugin_create_node(FuPlugin *plugin, FuDevice *device, GError **err
-		return FALSE;
+		mst_device = fu_dell_dock_mst_new(ctx);
+		if (fu_dell_dock_get_dock_type(FU_DEVICE(ec_device)) == DOCK_BASE_TYPE_ATOMIC)
+			instance_id = DELL_DOCK_VMM6210_INSTANCE_ID;
+		else
+			instance_id = DELL_DOCK_VM5331_INSTANCE_ID;
+		fu_device_add_instance_id(FU_DEVICE(mst_device), instance_id);
 
-	/* create package version endpoint */
-	status_device = fu_dell_dock_status_new(ctx);
-	if (fu_dell_dock_get_dock_type(FU_DEVICE(ec_device)) == DOCK_BASE_TYPE_ATOMIC)
-		instance_id_status = DELL_DOCK_ATOMIC_STATUS_INSTANCE_ID;
-	else if (fu_dell_dock_module_is_usb4(FU_DEVICE(ec_device)))
-		instance_id_status = DELL_DOCK_DOCK2_INSTANCE_ID;
-	else
-		instance_id_status = DELL_DOCK_DOCK1_INSTANCE_ID;
-	instance_guid_status = fwupd_guid_hash_string(instance_id_status);
-	fu_device_add_guid(FU_DEVICE(status_device), fwupd_guid_hash_string(instance_guid_status));
-	fu_device_add_child(FU_DEVICE(ec_device), FU_DEVICE(status_device));
-	fu_device_add_instance_id(FU_DEVICE(status_device), instance_id_status);
-	if (!fu_dell_dock_plugin_create_node(plugin, FU_DEVICE(status_device), error))
-		return FALSE;
+		instance_guid = fwupd_guid_hash_string(instance_id);
+		fu_device_add_guid(FU_DEVICE(mst_device), instance_guid);
+
+		if (!fu_device_probe(FU_DEVICE(mst_device), error))
+			return FALSE;
 
-	/* create TBT endpoint if Thunderbolt SKU and Thunderbolt link inactive */
-	if (fu_dell_dock_ec_needs_tbt(FU_DEVICE(ec_device))) {
-		g_autoptr(FuDellDockTbt) tbt_device = fu_dell_dock_tbt_new(proxy);
-		g_autofree const gchar *instance_guid_tbt =
-		    fwupd_guid_hash_string(DELL_DOCK_TBT_INSTANCE_ID);
-		fu_device_add_guid(FU_DEVICE(tbt_device), instance_guid_tbt);
-		fu_device_add_child(FU_DEVICE(ec_device), FU_DEVICE(tbt_device));
-		if (!fu_dell_dock_plugin_create_node(plugin, FU_DEVICE(tbt_device), error))
+		fu_device_add_child(FU_DEVICE(ec_device), FU_DEVICE(mst_device));
+		if (!fu_dell_dock_plugin_create_node(plugin, FU_DEVICE(mst_device), error))
 			return FALSE;
 	}
 
+	/* PACKAGE */
+	{
+		g_autoptr(FuDellDockStatus) status_device = NULL;
+		const gchar *instance_id = NULL;
+		g_autofree const gchar *instance_guid = NULL;
+
+		status_device = fu_dell_dock_status_new(ctx);
+		if (fu_dell_dock_get_dock_type(FU_DEVICE(ec_device)) == DOCK_BASE_TYPE_ATOMIC)
+			instance_id = DELL_DOCK_ATOMIC_STATUS_INSTANCE_ID;
+		else if (fu_dell_dock_module_is_usb4(FU_DEVICE(ec_device)))
+			instance_id = DELL_DOCK_DOCK2_INSTANCE_ID;
+		else
+			instance_id = DELL_DOCK_DOCK1_INSTANCE_ID;
+		fu_device_add_instance_id(FU_DEVICE(status_device), instance_id);
+
+		instance_guid = fwupd_guid_hash_string(instance_id);
+		fu_device_add_guid(FU_DEVICE(status_device), fwupd_guid_hash_string(instance_guid));
+
+		fu_device_add_child(FU_DEVICE(ec_device), FU_DEVICE(status_device));
+		if (!fu_dell_dock_plugin_create_node(plugin, FU_DEVICE(status_device), error))
+			return FALSE;
+	}
+
+	/* TBT 3 */
```

Comment:
This also covers TBT4 doesn't it?

Reply:
we do usb3 interface for TBT4, which should be available all the time.